fplll/bkz.cpp

```diff
@@ -310,16 +307,14 @@ bool BKZReduction<FT>::dsvp_postprocessing(int kappa, int block_size, const vect
 template <class FT>
 bool BKZReduction<FT>::svp_reduction(int kappa, int block_size, const BKZParam &par, bool dual)
 {
-  bool clean = true;
-
   int lll_start = (par.flags & BKZ_BOUNDED_LLL) ? kappa : 0;
-
-  if (!lll_obj.lll(lll_start, lll_start, kappa + block_size))
-  {
-    throw std::runtime_error(RED_STATUS_STR[lll_obj.status]);
-  }
-
-  clean &= (lll_obj.nSwaps == 0);
+  
+  int first = dual ? kappa + block_size - 1 : kappa;
+  
+  // ensure we are computing something sensible
+  m.updateGSORow(0, first);
+  FT old_first;
```

Comment:
you're right. we absolutely should you getRExp (to avoid overflows). i changed it. in the process i also replaced the gso update call with a size_reduction, since again i was noticing some weird numerical behavior (which only affected the clean flag in non-critical situations though).